embedx/config.schema.json

```
@@ -312,6 +312,225 @@
+              "examples": [
+                "https://path/to/metadata"
+              ]
+            },
+            "idp_certificate_path": {
+              "title": "IDP Certificate Path",
+              "description": "The path to the certificate of the IDP",
+              "type": "string",
+              "examples": [
+                "file://path/to/certificate",
+                "https://foo.bar.com/path/to/certificate"
+              ]
+            },
+            "idp_logout_url": {
+              "title": "IDP Logout URL",
+              "description": "The URL of the SLO API of the IDP",
+              "type": "string",
+              "examples": [
+                "https://path/to/logout"
+              ]
+            },
+            "idp_sso_url": {
+              "title": "IDP SSO URL",
+              "description": "The URL of the SSO Handler at the IDP",
+              "type": "string",
+              "examples": [
+                "https://path/to/sso"
+              ]
+            },
+            "idp_entity_id": {
+              "title": "The EntityID of the IDP",
+              "description": "It is a unique identifier representing the IDP in saml requests",
+              "type": "string",
+              "examples": [
+                "https://samltest.id/saml/idp"
+              ]
+            }
+          },
+          "allOf": [
+            {
+              "if": {
+                "properties": {
+                  "idp_metadata_url": {
+                    "const": {}
+                  }
+                }
+              },
+              "then": {
+                "required": [
+                  "idp_logout_url",
+                  "idp_certificate_path",
+                  "idp_entity_id"
+                ]
+              },
+              "else":{
+                "properties": {
+                  "idp_certificate_path": {
+                    "const": {}
+                  },
+                  "idp_logout_url": {
+                    "const": {}
+                  },
+                  "idp_entity_id":{
+                    "const":{}
+                  },
+                  "idp_sso_url":{
+                    "const":{}
+                  }
+                }
+              }
+            }
+          ]
+        },
+        "attributes_map": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "ID",
+              "description": "The name of the IDP attribute that will represent the user ID on Kratos",
+              "type": "string",
+              "examples": [
+                "file://path/to/oidc.jsonnet",
+                "https://foo.bar.com/path/to/oidc.jsonnet",
+                "base64://bG9jYWwgc3ViamVjdCA9I..."
+              ]
+            },
+            "firstname": {
+              "title": "Firstname",
+              "description": "The name of the IDP attribute that will represent the user's firstname on Kratos",
```

Here and for other attributes

```suggestion
              "description": "Name of the IDP attribute representing user firstname on Kratos",
```